src/resources/marketplace.js

```diff
@@ -289,36 +217,15 @@ class Marketplace extends Adaptable {
    * @return {Promise<{timestamp, transactionReceipt}>}
    */
   async addData(listingId, offerId, data, confirmationCallback) {
-    if (offerId) {
-      const { adapter, listingIndex, offerIndex } = this.parseOfferId(offerId)
-
-      // We expect this to be review data from the seller.
-      const ipfsHash = await this.reviewIpfsStore.save(data)
-      const ipfsBytes = this.contractService.getBytes32FromIpfsHash(ipfsHash)
-
-      return await adapter.addData(
-        ipfsBytes,
-        listingIndex,
-        offerIndex,
-        confirmationCallback
-      )
-    } else if (listingId) {
-      const { adapter, listingIndex } = this.parseListingId(listingId)
-
-      const ipfsHash = await this.listingIpfsStore.save(data)
-      const ipfsBytes = this.contractService.getBytes32FromIpfsHash(ipfsHash)
+    const ipfsHash = await this.reviewIpfsStore.save(data)
```

Comment:
The type of data in case it is an offer vs a listing is different.
For an offer it is review data, but for listing it is TBD (not being used yet).
So the validation and upload code to IPFS will need to be conditional.
Or perhaps we should split this method into addListingData and addOffer data ?

Reply:
Oh, I see. Yeah I like the idea of splitting into 2 separate methods; will save that for a separate PR though. Used if/else to fix this for now.